File: Leetcode 3501-3600/3536/3536.py
# Leetcode 3536: Maximum Product of Two Digits
# https://leetcode.com/problems/maximum-product-of-two-digits/
# Solved on 25th of July, 2026
class Solution:
    def maxProduct(self, n: int) -> int:
        """
        Finds the maximum product of two distinct digits of a given integer.
        @param n: The integer to find the maximum product of two digits.
        @return: The maximum product of two distinct digits.
        """
        firstMax = 0
        secondMax = 0

        while n > 0:
            currentDigit = n % 10
            if currentDigit > firstMax:
                secondMax = firstMax
                firstMax = currentDigit
            elif currentDigit > secondMax:
                secondMax = currentDigit
            n //= 10
        
        return firstMax * secondMax

File: Leetcode 3501-3600/3536/3536.java
// Leetcode 3536: Maximum Product of Two Digits
// https://leetcode.com/problems/maximum-product-of-two-digits/
// Solved on 25th of July, 2026
class Solution {
    /**
     * Finds the maximum product of two distinct digits of a given integer.
     * @param n The integer to find the maximum product of two digits.
     * @return The maximum product of two distinct digits.
     */
    public int maxProduct(int n) {
        int firstMax = 0;
        int secondMax = 0;
        while (n > 0) {
            int digit = n % 10;
            if (digit > firstMax) {
                secondMax = firstMax;
                firstMax = digit;
            } else if (digit > secondMax) {
                secondMax = digit;
            }
            n /= 10;
        }
        return firstMax * secondMax;
    }
}